apps/api/src/domain/games/memory-match/queries/memory-match.controller.ts

```diff
@@ -19,11 +23,28 @@ import { MemoryMatchService } from '../services/memory-match.service';
 export class MemoryMatchController {
     constructor(private readonly memoryMatchService: MemoryMatchService) {}
 
+    // TODO this should be a MemoryMatchRoundViewModel
```

Comment:
You need to define a class `MemoryMatchRoundViewModel` at `/models/memory-match-view-model.view-model.ts`. 
```ts
interface UrlBuilder{
     build(mediaItemId: string)
}
 
class MemoryMatchRoundViewModel{
    // add all properties here noting that ~~cardBackImageId~~ -> `cardbackImageUrl` and similar for all ID props

    constructor(round: MemoryMatchRound, private readonly urlBuilder: UrlBuilder){
        // ... move the logic from `MemoryMatchService.buildViewModel` here instead

   // you'll also need this
    if(isNonEmptyString(round.cardbackImageId){
         this.cardbackImageUrl = this.urlBuilder.build(round.cardbackImageId)
    }
    }
}